src/IncrementalChecker.ts

```diff
@@ -158,7 +176,14 @@ class IncrementalChecker {
       }
     }
 
-    this.program = IncrementalChecker.createProgram(this.programConfig, this.files, this.watcher, this.program);
+    if (this.hasVueLoader()) {
+      this.programConfig = this.programConfig || VueProgram.loadProgramConfig(this.programConfigFile);
```

Comment:
in this branch isn't it always the `VueProgram` you use?

Reply:
No no, this condition is here to detect if `vue-loader` is found in the user's project and then use the VueProgram if it is. We don't need to maintain a separate branch for Vue if we have this condition. It can be merged into master.

Reply:
`this.programConfig || VueProgram.loadProgramConfig(this.programConfigFile)` but wouldn't this statement always use `programConfig`? Or have I missed something?

Reply:
The first pass over that line `this.programConfig` will be undefined. The next time it passes over that line `this.programConfig` will have a value and it will grab that instead of performing another unnecessary `VueProgram.loadProgramConfig(...`.

Reply:
Cool - WFM!